.github/workflows/update-version.yml

```diff
@@ -0,0 +1,31 @@
+name: Update version file
```

Comment:
nice! ideally, the `binaries` workflow would depend on a successful run of this one. not sure how easy it is to implement that, i think it's fine the way it is

Reply:
@brunocalza maybe this would work? https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run

```
name: Generate binaries
on:
  workflow_run:
    workflows: ["Update version file"]
    types:
      - completed
```

i have the `update-version` workflow to only work on `main`, though, so would that pose an issue if the generated binaries workflow was dependent on it?

Reply:
hum, I missed this was triggered when a push happens to `main`. Ideally, it would be triggered when a release is created, right? and the `binaries` would be run after that. because if you create `version.json` when a push happens, the next binary release will have the wrong version

Reply:
@brunocalza ah, i see. when the [binaries are generated](https://github.com/tablelandnetwork/basin-cli/actions/runs/7213279552/job/19652803514#step:6:1), it looks like that's where the tag gets created, which is what i fetch to get the version in the CLI. 
```
  • publishing
    • scm releases
      • creating or updating release                 tag=v0.0.6 repo=tablelandnetwork/basin-cli
      • release updated                              name=v0.0.6 release-id=133993550 request-id=3BC0:2FB0:182D22B:1919694:657B5041
      • uploading to release                         file=dist/basin-darwin-amd64.tar.gz name=basin-darwin-amd64.tar.gz
      • uploading to release                         file=dist/basin-linux-arm64.tar.gz name=basin-linux-arm64.tar.gz
      • uploading to release                         file=dist/basin-linux-amd64.tar.gz name=basin-linux-amd64.tar.gz
      • uploading to release                         file=dist/basin-darwin-arm64.tar.gz name=basin-darwin-arm64.tar.gz
      • published                                    url=https://github.com/tablelandnetwork/basin-cli/releases/tag/v0.0.6
      • took: 4s
```

hm, so maybe my approach won't work because i'd need my GH action to run after that tag is created but before the binaries are generated?

Reply:
The tag is not being created by the workflow. The tag is created by Github itself when a release is created. 

I think you need to switch to 
```
on:
  release:
    types:
      - created
```
 and somehow get access to the tag being created inside the workflow. 

You can test that, by [drafting new releases](https://github.com/tablelandnetwork/basin-cli/releases/new) targeting your branch.  

If this is taking too much of your time. Feel free to merge it and I can figure that out for you.

Reply:
👍 feel free to merge it